celements-scheduler/src/main/java/com/celements/scheduler/job/AbstractJob.java

```diff
@@ -100,94 +91,49 @@ public final void execute(JobExecutionContext jobContext) throws JobExecutionExc
-    ec.setProperty("xwikicontext", scontext);
-
-    ecim.initialize(ec);
-    execution.setContext(ec);
-
-    setupServerUrlAndFactory(scontext, xwikiContext);
-    VelocityManager velocityManager = Utils.getComponent(VelocityManager.class);
-    velocityManager.getVelocityContext();
-    return execution;
+  ExecutionContext initExecutionContext(JobDataMap data) throws ExecutionContextException {
+    ExecutionContext context = createEContextForJob(data);
+    executionContextManager.get().initialize(context);
+    prepareXContextForJob(data);
+    return context;
   }
 
   /**
-   * create a new copy of the xwiki context. Job Executions always use a different thread.
-   * The xwiki context is NOT thread safe and may not be shared.
-   * TODO CELDEV-534
-   *
-   * @throws DocumentNotExistsException
-   * @throws MalformedURLException
+   * create the execution context with job defaults to allow proper initialization by the
+   * ExecutionContextManager.
    */
-  XWikiContext createJobContext(XWikiContext xwikiContext) {
-    final XWiki xwiki = xwikiContext.getWiki();
-    final String database = xwikiContext.getDatabase();
-
-    // We are sure the context request is a real servlet request
-    // So we force the dummy request with the current host
-    XWikiServletRequestStub dummy = new XWikiServletRequestStub();
-    dummy.setHost(xwikiContext.getRequest().getHeader("x-forwarded-host"));
-    dummy.setScheme(xwikiContext.getRequest().getScheme());
-    XWikiServletRequest request = new XWikiServletRequest(dummy);
-
-    // Force forged context response to a stub response, since the current context response
-    // will not mean anything anymore when running in the scheduler's thread, and can cause
-    // errors.
-    XWikiResponse response = new XWikiServletResponseStub();
-
-    // IMPORTANT: do NOT clone xwikiContext. You would need to ensure that no reference or
-    // unwanted value leaks in the new context.
-    // IMPORTANT: following lines base on XWikiRequestInitializer.prepareContext
-    XWikiContext scontext = new XWikiContext();
-    scontext.setEngineContext(xwikiContext.getEngineContext());
-    scontext.setRequest(request);
-    scontext.setResponse(response);
-    scontext.setAction("view");
-    scontext.setDatabase(database);
-
-    // feed the job context
-    scontext.setUser(xwikiContext.getUser());
-    scontext.setLanguage(xwikiContext.getLanguage());
-    scontext.setMainXWiki(xwikiContext.getMainXWiki());
-    scontext.setMode(XWikiContext.MODE_SERVLET);
-
-    scontext.setWiki(xwiki);
-    scontext.getWiki().getStore().cleanUp(scontext);
-
-    scontext.flushClassCache();
-    scontext.flushArchiveCache();
-    return scontext;
+  private ExecutionContext createEContextForJob(JobDataMap data) {
+    ExecutionContext context = new ExecutionContext();
+    execution.get().setContext(context);
+    XWikiDocument jobDoc = ((XWikiDocument) data.get("jobDoc")).clone();
```

Comment:
only set the jobDocRef in the data map and fetch it in the job-thread instead of cloning.